_user/rotrectangle/main.c

```diff
@@ -210,32 +201,34 @@ static int rotrectangle_startpanel(rotrectangle_t *self)
 	printf("    )'q' - exit\n");
 	printf("  Press enter to start: \n");
 	getchar();
-
-	return 0;
 }
 
 
 int main(void)
 {
 	int ret;
+	unsigned int stacksz;
 	char *stack;
 
-	stack = (char *)malloc(1024 * sizeof(char));
-
 	graph_adapter_t adapter = GRAPH_ANY;
 	graph_mode_t mode = GRAPH_DEFMODE;
 	graph_freq_t freq = GRAPH_DEFFREQ;
 	graph_t graph;
 
 	rotrectangle_t rotrectangle;
 
+	stacksz = 1024;
+
+	if ((stack = (char *)malloc(stacksz)) == NULL)
```

Comment:
Casting malloc result is a bad practice:

https://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc

Reply:
Alek, I don't understand your remark. This line is correct and such construction is widely used in Phoenix-RTOS code.

Reply:
There is no need to cast explicitly `void *` to any other type. Some say it's a bad practice, other say otherwise. If you think this is ok, I won't oppose.